components/eventing-controller/upgrade-job/process/delete_beb_subscriptions.go

```diff
@@ -0,0 +1,93 @@
+package process
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/avast/retry-go"
+	pkgerrors "github.com/pkg/errors"
+)
+
+var _ Step = &DeleteBebSubscriptions{}
+
+// DeleteBebSubscriptions struct implements the interface Step
+type DeleteBebSubscriptions struct {
+	name    string
+	process *Process
+}
+
+// NewDeleteBebSubscriptions returns new instance of NewDeleteBebSubscriptions struct
+func NewDeleteBebSubscriptions(p *Process) DeleteBebSubscriptions {
+	return DeleteBebSubscriptions{
+		name:    "Delete BEB subscriptions from Event-Mesh",
+		process: p,
+	}
+}
+
+// ToString returns step name
+func (s DeleteBebSubscriptions) ToString() string {
+	return s.name
+}
+
+// Do deletes the subscriptions from Event Mesh (BEB)
+func (s DeleteBebSubscriptions) Do() error {
+	// First check if BEB client is initialized or not
+	if !s.process.Clients.EventMesh.IsInitialised() {
+		return errors.New("event mesh (BEB) client is not initialised")
+	}
+
+	// Traverse through the subscriptions and migrate
+	subscriptionListItems := s.process.State.FilteredSubscriptions.Items
+
+	for _, subscription := range subscriptionListItems {
+		s.process.Logger.WithContext().Info("Deleting Event Mesh (BEB) Subscription: ", subscription.Name)
+		err := s.DeleteBEBSubscription(subscription.Name)
+		if err != nil {
+			s.process.Logger.WithContext().Error(err)
+			continue
```

Comment:
@marcobebway Please check that should we continue if we fail to delete a single BEB subscription from the list? or return error and fail the upgrade job?

Reply:
updated, we are returning the error.